pkg/license/download.go

```diff
@@ -0,0 +1,253 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package license
+
+import (
+	"crypto/sha1"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"path/filepath"
+
+	"github.com/nozzle/throttler"
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+	"k8s.io/release/pkg/util"
+)
+
+// ListURL is the json list of all spdx licenses
+const ListURL = "https://spdx.org/licenses/licenses.json"
+
+// NewDownloader returns a downloader with the default options
+func NewDownloader() (*Downloader, error) {
+	return NewDownloaderWithOptions(DefaultDownloaderOpts)
+}
+
+// NewDownloaderWithOptions returns a downloader with specific options
+func NewDownloaderWithOptions(opts *DownloaderOptions) (*Downloader, error) {
+	if err := opts.Validate(); err != nil {
+		return nil, errors.Wrap(err, "validating downloader options")
+	}
+	impl := DefaultDownloaderImpl{}
+	impl.SetOptions(opts)
+
+	d := &Downloader{}
+	d.SetImplementation(&impl)
+
+	return d, nil
+}
+
+// DownloaderOptions is a set of options for the license downloader
+type DownloaderOptions struct {
+	CacheData         bool   // Should we use the cache or not
```

Comment:
The name of the flag seems misleading to me, let's indicate that it enables something. How about:
```suggestion
	EnableCache         bool   // Should we use the cache or not
```

Reply:
It's funny that I did not think of changing it even after getting confused myself several times during testing. 
Changed :+1: